test/dialogs/proposal_test.rb

```diff
@@ -32,12 +32,20 @@
 
   describe "#run" do
```

Comment:
I miss some test here checking for collapsing-expanding actions. Just something that tests that those actions are delegated to the widget should be enough.